crates/dekoder/src/inventory.rs

```diff
@@ -0,0 +1,86 @@
+//! Utilities for dealing with the eko assets.
+use lz4_flex::frame::FrameDecoder;
+use ndarray_npy::NpzReader;
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::fs::{read_dir, read_to_string, File};
+use std::io::Cursor;
+use std::path::PathBuf;
+use yaml_rust2::{Yaml, YamlLoader};
+
+use crate::{EKOError, Operator, Result};
+
+/// Headers are in yaml files.
+const HEADER_EXT: &str = "yaml";
+
+/// Assets manager.
+pub(crate) struct Inventory<K: Eq + for<'a> TryFrom<&'a Yaml, Error = EKOError>> {
+    /// Working directory
+    pub(crate) path: PathBuf,
+    /// Available keys
+    pub(crate) keys: HashMap<OsString, K>,
+}
+
+impl<K: Eq + for<'a> TryFrom<&'a Yaml, Error = EKOError>> Inventory<K> {
+    /// Load all available entries.
+    pub fn load_keys(&mut self) -> Result<()> {
+        for entry in read_dir(&self.path)? {
+            // is header file?
+            let entry = entry?.path();
+            if !entry.extension().is_some_and(|ext| ext.eq(HEADER_EXT)) {
```

Comment:
Use `ext == HEADER_EXT` instead of `ext.eq(HEADER_EXT)`.

Reply:
d394978d